### PR TITLE
Add search for desktop and mobile

### PR DESCRIPTION
**Pull request by lukeocodes**

### CODE REVIEW

File: shared/src/shared/components/search/DocSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		searchButtonM,
+	});
+
+	if (!isOpen) return null;
+
+	return createPortal(
+		<DocSearchModal
+			initialQuery={initialQuery}
+			initialScrollY={window.scrollY}
+			onClose={onClose}
```
Comment (SandraRodgers) [inline shared/src/shared/components/search/DocSearch.tsx:51]: Do you want to leave these comments in?

**Review by SandraRodgers — APPROVED** [review]

File: shared/src/shared/components/layout/Header.astro [hunk 2]
```diff
@@ -96,7 +94,7 @@ import MobileNavItem from "../nav/MobileNavItem.astro";
 						</Link>
 					</div>
 					<div class="flex items-center space-x-5 xl:ml-12 pb-1 xl:hidden">
-						<Button icon="search" class="button big-icon button--main-nav" aria-label="Search" />
+						<Button id="docsearch-button-m" icon="search" class="button big-icon button--main-nav" aria-label="Search" />
```
Comment (briancbarrow) [inline shared/src/shared/components/layout/Header.astro:97]: I'm assuming these indicate mobile and desktop? Any reason we don't want to spell it out? It'll make future debugging easier I think

**Review by briancbarrow — APPROVED** [review]

File: shared/src/shared/components/layout/Header.astro (hunk 2 above)
Comment (lukeocodes) [reply shared/src/shared/components/layout/Header.astro:97]: That is a fair point! I will change it, thank you

File: shared/src/shared/components/search/DocSearch.tsx (hunk 1 above)
Comment (lukeocodes) [reply shared/src/shared/components/search/DocSearch.tsx:51]: Yup for now

**Pull request merged**